### ISSUE TITLE
I hate how the API for downloading files work.

### ISSUE THREAD

**Issue by wjaskowski** [neptune-ai/neptune-client#725]
I have written already #724 but let me add here a few bits.

Let's say I have a file `myresults.csv` and I upload it:
```python
run['my/predictions/results'].upload('/tmp/myresults.csv')
```
What happened? Now under the key `predictions/results,` we have a file called `results.csv`. So the original name is gone but the original extension is preserved. What is the logic behind it? No idea.

Now, I would like to download it. If I know the name of the key, this is very easy:
```python
run['my/predictions/results'].download()
```
But if I don't, it becomes very inconvenient. What one would normally do is to list all the files under `predictions` but this doesn't work:
```python
run['my/predictions'].fetch()
```
outputs all the ordinary fields but not the files, so it is `{}` in this case.

To get the files I would need to use `run.get_structure()['my']['predictions']`, which outputs:
```
{'results': <neptune.new.attributes.atoms.file.File at 0x7fb7470fc7f0>}
```
But better if 'my' and 'predictions' are literals because `run.get_structure()['my.predictions']` won't work, obviously.

Now, I know the name of my file, don't I? No, I don't actually, I know just the key. It is enough to download the file to some directory e.g., `run['my/predictions/results'].download()` but it is not enough to open it since I don't know the extension. To get it I can either list the content of the destination directory or call:
```python
run.get_structure()['my']['predictions']['results'].fetch_extension()
```
Now it is enough to join the key and the extension, and voila - I can eventually open the downloaded file and start doing the actual work.

Yes, I got a bit irritated, indeed. Sorry about that but I just spent on it significantly too much time. The design of this API is suboptimal. Maybe there is just a simple way but I just cannot find it.

**Comment by pkasprzyk** [comment]
@wjaskowski two things:
1. Most of our examples omit file extension from the attribute name, because most of the time the extension of file you upload is obvious and it doesn't make sense to duplicate data. But if your use case is different, nothing stops you from doing `run['my/predictions/results.csv'].upload('/tmp/myresults.csv')` instead of `run['my/predictions/results'].upload('/tmp/myresults.csv')`.
2. I'm not sure what's your use case. Are doing this download interactively (in this case `run.print_structure()` can be of help, among others), or in a script?

**Comment by wjaskowski** [comment]
> Nothing stops you from doing run['my/predictions/results.csv'].upload('/tmp/myresults.csv') instead of run['my/predictions/results'].upload('/tmp/myresults.csv').

Yes. If I knew in advance that this will be a problem at some point then I would organize things in this way, indeed but now is too late because I have already too many runs in the project.

Anyway, maybe it would be useful to somebody. I now use the following workaround to get the extension for any key given:

```python
def index_hierarchical_dict(d: Dict, key: str):
    """Returns d[key[0]][key[1]]...[key[-1]]"""
    key.split("/")
    for k in key.split("/"):
        d = d[k]
    return d


def get_filename_extension_for_key(neptune_run: Run, key: str) -> str:
    """Returns extension e.g. 'txt' for a given key"""
    struct_dict = neptune_run.get_structure()
    file_obj = index_hierarchical_dict(struct_dict, key)
    return file_obj.fetch_extension()
```

**Comment by elgehelge** [comment]
For others with similar needs:
There also exists a field type called FileSet. Which represent a bunch of files, each file keeping there relative folder location and file extension. I believe you can only download the whole set at once, so in this case, you would need a FileSet for your results, and another FileSet for eg. visualizations of samples.
